### PR TITLE
Rework capture-build-details to allow for overriding using WORKSPACE_ROOT

### PR DESCRIPTION
**Pull request by rickpasetto**
Also cleaned it up to add a usage message, and allow for passing in other parameters.

### CODE REVIEW

File: Scripts/capture-build-details.sh [hunk 1]
```diff
@@ -5,27 +5,78 @@
...
+  echo "INFO: ${*}" >&2
+}
+
+git_source_root="${WORKSPACE_ROOT:-${SRCROOT}}"
 plist="${BUILT_PRODUCTS_DIR}/${INFOPLIST_PATH}"
+profile_path="${HOME}/Library/MobileDevice/Provisioning Profiles/${EXPANDED_PROVISIONING_PROFILE}.mobileprovision"
+xcode_build_version=${XCODE_PRODUCT_BUILD_VERSION:-$(xcodebuild -version | grep version | cut -d ' ' -f 3)}
+while [[ $# -gt 0 ]]
+do
+  arg="$1"
```
Comment (darinkrauss) [inline Scripts/capture-build-details.sh]: [nit] Separate `arg` local variable is unnecessary. Can just `case $1 in`, I believe.

File: Scripts/capture-build-details.sh [hunk 2]
```diff
@@ -5,27 +5,78 @@
...
+
+  case $arg in
+    -r|--git-source-root)
+      git_source_root="${2}"
+      shift 2
+      ;;
+    -i|--info-plist-path)
+      plist="${2}"
```
Comment (darinkrauss) [inline Scripts/capture-build-details.sh]: [nit] Suggest `plist` -> `info-plist-path` for consistency.

File: Scripts/capture-build-details.sh [hunk 3]
```diff
@@ -5,27 +5,78 @@
...
+      shift 2
+      ;;
+    -p|--provisioning-profile-path)
+      profile_path="${2}"
```
Comment (darinkrauss) [inline Scripts/capture-build-details.sh]: [nit] Suggest `profile-path` -> `provisioning-profile-path` for consistency.

File: Scripts/capture-build-details.sh [hunk 4]
```diff
@@ -5,27 +5,78 @@
...
+      shift 2
+      ;;
+  esac
+done
 
+if [ ${#} -ne 0 ]; then
+  error "Unexpected arguments: ${*}"
+fi
+
+if [ "${git_source_root}" == "" ]; then
```
Comment (darinkrauss) [inline Scripts/capture-build-details.sh]: [nit] `-n`?

File: Scripts/capture-build-details.sh [hunk 5]
```diff
@@ -5,27 +5,78 @@
...
-cd "${SRCROOT}"
+SCRIPT="$(basename "${0}")"
+SCRIPT_DIRECTORY="$(dirname "${0}")"
 
+error() {
+  echo "ERROR: ${*}" >&2
+  echo "Usage: ${SCRIPT} [-r|--git-source-root git-source-root] [-p|--provisioning-profile-path provisioning-profile-path] [-i|--info-plist-path info-plist-path]" >&2
+  echo "Parameters:" >&2
+  echo "  -r|--git-source-root <git-source-root> root location of git repository to gather information from; optional, defaults to \$WORKSPACE_ROOT if present, otherwise defaults to \$SRCROOT" >&2
+  echo "  -p|--provisioning-profile-path         path to the .mobileprovision provisioning profile file to check for expiration; optional, defaults to \${HOME}/Library/MobileDevice/Provisioning Profiles/\${EXPANDED_PROVISIONING_PROFILE}.mobileprovision" >&2
```
Comment (darinkrauss) [inline Scripts/capture-build-details.sh]: [nit] Missing parameter `<provisioning-profile-path>` here and `<info-plist-path>` on the line below.

File: Scripts/capture-build-details.sh [hunk 6]
```diff
@@ -5,27 +5,78 @@
...
+      ;;
+  esac
+done
 
+if [ ${#} -ne 0 ]; then
+  error "Unexpected arguments: ${*}"
+fi
+
+if [ "${git_source_root}" == "" ]; then
+  error "Must provide one of --git-source-root, \$WORKSPACE_ROOT, or \$SRCROOT"
```
Comment (darinkrauss) [inline Scripts/capture-build-details.sh]: [nit] Suggest using squiggly brackets here for consistency with the rest of the file.

File: Scripts/capture-build-details.sh [hunk 7]
```diff
@@ -5,27 +5,78 @@
...
+info "Gathering build details in ${git_source_root}"
+cd "${git_source_root}"
 
 if [ -e .git ]; then
   rev=$(git rev-parse HEAD)
   plutil -replace com-loopkit-Loop-git-revision -string ${rev} "${plist}"
-  branch=$(git branch | grep \* | cut -d ' ' -f2-)
-  plutil -replace com-loopkit-Loop-git-branch -string "${branch}" "${plist}"
+  branch=$(git branch --show-current)
+  if [ "$branch" != "" ]; then
```
Comment (darinkrauss) [inline Scripts/capture-build-details.sh]: [nit] `-z`?

**Review by darinkrauss — APPROVED** [review]
A number of nits for your consideration, but otherwise LGTM! ⛰️

File: Scripts/capture-build-details.sh (hunk 2 above)
Comment (rickpasetto) [reply Scripts/capture-build-details.sh]: 👍 I'll use `info_plist_path`

File: Scripts/capture-build-details.sh (hunk 3 above)
Comment (rickpasetto) [reply Scripts/capture-build-details.sh]: 👍 I'll use `provisioning_profile_path`

File: Scripts/capture-build-details.sh (hunk 4 above)
Comment (rickpasetto) [reply Scripts/capture-build-details.sh]: Sure, but I presume you meant `-z` right?

File: Scripts/capture-build-details.sh (hunk 7 above)
Comment (rickpasetto) [reply Scripts/capture-build-details.sh]: sure, but `-n`

File: Scripts/capture-build-details.sh (hunk 4 above)
Comment (darinkrauss) [reply Scripts/capture-build-details.sh]: Yea, that's what I meant. 😉

**Pull request merged**